portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -23,42 +23,80 @@
 import javax.servlet.http.HttpServletResponse;
 import java.util.List; 
 import java.util.ArrayList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/comments-data")
 public class CommentsServlet extends HttpServlet {
-    private List<Comment> comments;
-
-    @Override
-    public void init(){
-        comments = new ArrayList<Comment>();
-    }
-
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         Comment newComment = getCommentFromRequest(request);
-        comments.add(newComment);
+        putCommentToDatastore(newComment);
         response.sendRedirect("/contact.html");
     }
 
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{ 
+        PreparedQuery results = getCommentsFromDatastore(); 
+        List<Comment> comments = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            comments.add(getCommentFromCommentEntity(entity));
+        }
+        response.setContentType("application/json;");
```

Comment:
I'm not sure if it will work, but I wonder if we could pass `results.asIterable()` directly into `convertToJsonUsingGson()` without having to manually convert to an `ArrayList`.

Reply:
done with results.asList()

Reply:
NIce!